pagopa/io-app
# [IOBP-563, IOBP-612] Add empty states and navigation events to `PaymentsHomeScreen`

> [!WARNING]
> This PR depends on https://github.com/pagopa/io-app/pull/5659

## Short description
This PR introduces empty states and navigation events within the new payments landing screen

## List of changes proposed in this pull request
- Simplified `PaymentCardSmall` and `PaymentCardCarousel` components
- Added empty states to `PaymentHomeScreen`
- Added store reducer, actions and migration to manage empty methods banner visibility persistence
- Added navigation events to `PaymentHomeScreen` components
- Added unit tests
- Added missing locale keys
- Added payload to `paymentsDeleteMethodAction.success` in order to remove the wallet from the list once deleted successfully 

## How to test
Static check should solve correctly.
With the `io-dev-api-server`, enable te **New wallet section** from the **Profile** screen. 
Try to simulate different states by mocking api-server responses (`GET /ecommerce/wallets` and `GET /wallet/v1/transactions`). Check that the screen is rendered like in the previews below.

## Preview

|                      | **w/o payment methods** | **w/ payment methods** |
|----------------------|--------------------------|-------------------------|
| **w/o transactions** |  <img src="https://github.com/pagopa/io-app/assets/6160324/58057e3e-ef51-4f02-bc3a-4bbbf24cc199" width="150" /> |          <img src="https://github.com/pagopa/io-app/assets/6160324/afd1c408-5d9a-48a6-a13e-e5967e2817a2" width="150" />                |
| **w/ transactions**  |           <img src="https://github.com/pagopa/io-app/assets/6160324/b4c5e6b8-3b7e-42fc-b6a5-03f8a907616d" width="150" />                |            <img src="https://github.com/pagopa/io-app/assets/6160324/9881b73f-a7b2-415c-b3ca-512ee80a3ab0" width="150" />              |
